### PR TITLE
Adding in 'Strip Ship' functionality.

### PR DESCRIPTION
**Pull request by shearn89**
Something that was mentioned on reddit I believe!

This commit adds a simple button next to the save/reload icons that strips the ship to maximum class, D-rated modules, and no optional modules. Still needs a custom icon! May try to add in future things like 'all cargo' or 'fill empty with...' options.

### CODE REVIEW

**Comment by shearn89** [comment]
First time playing with angular, so refactoring may be required - definitely just hacked this together!

**Comment by cmmcleod** [comment]
This is pretty spot on. Well done :)

I was thinking about using a feather icon for lightest build. Thoughts? 

I'll gather the necessary icons

File: app/views/page-outfit.html [hunk 1]
```diff
@@ -3,6 +3,9 @@
   <div id="overview">
     <h1 ng-bind="ship.name"></h1>
     <div id="build">
+      <button ng-click="stripBuild()">
+        <svg class="icon lg"><use xlink:href="#spinner11"></use></svg><span class="button-lbl">Strip</span>
+      </button>
```
Comment (cmmcleod) [inline app/views/page-outfit.html]: I'd put the button at the end of the #build div, Such that the flow is:

[Build name field] [Save] [Reload] [Delete] [Reset] [Strip]

Comment (shearn89) [inline app/views/page-outfit.html]: Makes sense!

**Comment by shearn89** [comment]
Feather sounds good - I was trying to think of what would work but the feather is intuitive. Glad you like it!

File: app/js/controllers/controller-outfit.js [hunk 2]
```diff
@@ -117,6 +117,22 @@ angular.module('app').controller('OutfitController', ['$window', '$rootScope', '
   };
 
   /**
+   * Strip ship to D-class and no other components.
+   */
+  $scope.stripBuild = function() {
+		angular.forEach(ship.common, function(slot,i) {
```
Comment (cmmcleod) [inline app/js/controllers/controller-outfit.js]: Minor thing, but the build will fail unless you use spaces instead of tabs.
2 spaces per tab :)

Comment (shearn89) [inline app/js/controllers/controller-outfit.js]: Ah - that'll be my vim settings - fixed.

**Comment by cmmcleod** [comment]
:+1:

**Pull request merged**